portlist/portlist_macos.go

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin,amd64
+// +build darwin,amd64 go1.16,darwin,arm64
```

Comment:
this doesn't look right?

Reply:
Hmmm, may I ask why? This is to include this file when building native Apple M1 binaries.

Removing it results in a compile error using `go version go1.16beta1 darwin/arm64`.

Reply:
Doesn't `a,b` already include `a,b,c`?

> when building native Apple M1 binaries

Perhaps some of your other build tags are wrong and that's the conflict?

But I don't see how this makes a difference. This file looks like it'll be included with Go 1.15 or Go 1.16 on macOS with or without the new clause.

Reply:
Thanks for taking the time to explain!

> Doesn't `a,b` already include `a,b,c`?

Please note that the first build option refers to `amd64`, while the second refers to `arm64`. Could it be that the source of confusion?

> But I don't see how this makes a difference. This file looks like it'll be included with Go 1.15 or Go 1.16 on macOS with or without the new clause.

How I read it. First build option says to include the file when building for **darwin** and **amd64** (so macOS, Intel silicon), as it was originally. The second build option says when go1.16 include this file as well when building for **darwin** and **arm64** (so macOS, Apple silicon). Pre1.16 we don't want to include this file because **darwin,arm64** means iOS. But I might be misunderstanding  something.

Reply:
Argh, sorry! This is only the 74th time I've misread arm64 and amd64 as being the same. You'd think I'd learn by now.